js/app/cookies-banner.js

```diff
@@ -41,6 +56,17 @@ function extractDomainFromUrl(url) {
     return "." + secondLevelDomain + topLevelDomain;
 }
 
-$(function() {
-    initCookiesBanner();
-});
\ No newline at end of file
+function hasCookiesPreferencesSet() {
+    var cookies = document.cookie.split(";");
+
+    cookies = cookies.map(function (el) {
+        return el.trim();
+    });
+
+    for (var i = 0; i < cookies.length; i++) {
```

Comment:
```
for (var i = 0; i < cookies.length; i++) { 
        if (cookies[i] === "cookies_preferences_set=true") {
            return true;
        }
    }
```
This is fine, just a note the same can be done in a more concise way without losing clarity by using
```
if (cookies.indexOf("cookies_preferences_set=true") > 0) { 
    return true
}
```
Or a step further 
```
return cookies.indexOf("cookies_preferences_set=true") > 0
```
When using indexOf you can also use a bitwise operator but I do not recommend using this as clarity can be lost. But, here is an example of it in use:
```
if  (~cookies.IndexOf("cookies_preferences_set=true")){...
```
Note how the `~` (bitwise not) will return a negative integer if found but a boolean comparison (which the if statement will make) on a negative number will yield `true` as a value is present but a bitwise not on `-1` yields a `0` which when with a boolean comparison will yield a `false`. An easier way to remember the bitwise not operator is that it basically does this: `(value+1)*-1`

IndexOf also works on strings! With that in mind, a more concise way of writing this function might have been just to do

```
return ~document.cookie.indexOf("cookies_preferences_set=true") ? true : false
```

But a less fun but more readable and shorter way would be:

```
return document.cookie.indexOf("cookies_preferences_set=true") > -1
```

Or an even more readable and no fun way would be:
```
var isPreferenceSet = document.cookie.indexOf("cookies_preferences_set=true") > -1
return isPreferenceSet
```

I wish I got to do more with JavaScript 😞.

Reply:
The info about the bitwise operator was new to me, so thanks! Agree that using `indexOf` is a good balance between clarity and conciseness